config/github/src/main/kotlin/GitHubConfigFileProvider.kt

```diff
@@ -154,6 +180,17 @@ class GitHubConfigFileProvider(
             }
         }
 
+        /**
+         * Create the cache for configuration data based on the given [config].
+         */
+        private fun createCache(config: Config): GitHubConfigCache =
+            config.getStringOrNull(CACHE_DIRECTORY)?.let { cacheDir ->
+                logger.debug("Using file-based cache in directory '{}'.", cacheDir)
+
+                val lockCheckInterval = config.getInt(LOCK_CHECK_INTERVAL)
```

Comment:
Can you have a suggested default value for this property ?

Reply:
A default value is set in the `application.conf` files.

Reply:
I saw but I don't feel comfortable having the default value only in a `.conf` file. If anyone instantiate your class without it, it will crash.
Just a gut feeling, I won't block the PR over it.

Reply:
Another question out of curiosity: `GitHubConfigFileCache` expects the `lockCheckInterval` to be a `duration`.
But you read it with `config.getInt`. How does it work ?

Reply:
A conversion to a duration is done via the `lockCheckInterval.seconds` extension property.

Reply:
> I saw but I don't feel comfortable having the default value only in a `.conf` file. If anyone instantiate your class without it, it will crash. Just a gut feeling, I won't block the PR over it.

I made the cache implementations *internal*; then they cannot be instantiated by clients.